R/tbl.R

```diff
@@ -1,39 +1,211 @@
+#' Create a local lazy tbl
+#'
+#' Useful for testing KQL generation without a remote connection.
+#'
+#' @keywords internal
+#' @export
+#' @examples
+#' library(dplyr)
+#' df <- data.frame(x = 1, y = 2)
+#'
+#' df <- tbl_abstract(df, src = simulate_kusto())
+#' df %>% summarise(x = sd(x)) %>% show_query()
+tbl_abstract <- function(df, table_name, src = NULL) {
```

Comment:
Consider renaming the class to `tbl_kusto_abstract`, current name is very generic and may clash with other packages